app/addons/components/react-components.react.jsx

```diff
@@ -60,6 +62,101 @@ function (app, FauxtonAPI, React, Stores, FauxtonComponents, ace, beautifyHelper
     }
   });
 
+
+  var BulkActionComponent = React.createClass({
+
+    propTypes: {
+      hasSelectedItem: React.PropTypes.bool.isRequired,
+      removeItem: React.PropTypes.func.isRequired,
+      selectAll: React.PropTypes.func.isRequired,
+      toggleSelect: React.PropTypes.func.isRequired,
+      isChecked: React.PropTypes.bool.isRequired
+    },
+
+    getDefaultProps: function () {
+      return {
+        disabled: false,
+        title: 'Select rows that can be...',
+        bulkIcon: 'fonticon-trash',
+        buttonTitle: 'Delete all selected',
+        dropdownContentText: 'Deleted',
+        enableOverlay: false
+      };
+    },
+
+    render: function () {
+      return (
+        <div className="bulk-action-component">
+          <div className="bulk-action-component-selector-group">
+            {this.getMasterSelector()}
+            {this.getMultiSelectOptions()}
+          </div>
+        </div>
+      );
+    },
+
+    getMultiSelectOptions: function () {
+      if (!this.props.hasSelectedItem) {
+        return null;
+      }
+
+      return (
+        <button
+          onClick={this.props.removeItem}
+          className={'fonticon ' + this.props.bulkIcon}
```

Comment:
hm, we have no rule that we have to use double quotes in a computed property. i usually go with single quotes as it is a computed one, and not declarative markup